### ISSUE TITLE
Advice on using SPATA2 for Xenium data

### ISSUE THREAD

**Issue by HarryZhang1224** [theMILOlab/SPATA2#199]
Hi Jan,  

I am using SPATA2 on a Xenium dataset and used createGroupAnnotations to create the spatial annotation. However, I noticed that the outlines do not capture all the groups I want. In the image below I have pointed out in blue 3 examples of where the annotation missed the points/groups. Can you please give some advice on how to best approach this?

This is similar to [issues #165 ](https://github.com/theMILOlab/SPATA2/issues/165), but the solution did not work for me.

Thank you!

<img width="685" alt="Screenshot 2024-11-10 at 20 43 35" src="https://github.com/user-attachments/assets/a281ecc1-c4de-4f11-bfa3-5407d5a186b8">

My code for creating the annotation and plotting:
```r
obj <- qread(glue("/lila/data/deyk/harry/spatial/RA_Xenium/data/spata2_objs/spata2.qs"))
obj_seurat <- qread(glue("/data/deyk/harry/spatial/RA_Xenium/data/post_qc_data_baysor/annotated_QCed_seurat_obj.qs"))
lining_signature <- c("CLIC5", "MMP3", "MMP1", "HTRA4", "ITGB8")
obj_seurat <- AddModuleScore(obj_seurat, name = "lining", features=list("lining"=lining_signature), ctrl=20, nbin=24)
obj_seurat@meta.data %>% mutate(barcodes=rownames(.), liningFLS=as.factor(ifelse(lining1>0 & grepl("^FC", celltype_subcluster), "1", "0"))) -> obj_seurat@meta.data
obj <- addMetaDataObs(obj, obj_seurat@meta.data %>% dplyr::select(barcodes, lining1, liningFLS))
obj <- createGroupAnnotations(
    object = obj, 
    grouping = "liningFLS", 
    group = "1",
    id = "liningFLS", 
    force1 = FALSE,
    overwrite = TRUE,
    use_dbscan=FALSE,
    min_size=0
)
cluster_lining_FLS_outline <- ggpLayerSpatAnnOutline(obj, ids=getSpatAnnIds(obj))
plotSurface(obj, color_by="liningFLS", outline=TRUE) + 
  cluster_lining_FLS_outline
```

**Comment by kueckelj** [comment]
Hi, do you want to annotate what is displayed as brown in the upper plot?

**Comment by HarryZhang1224** [comment]
> Hi, do you want to annotate what is displayed as brown in the upper plot?

Yes, basically I want the annotation to circle region corresponding to the brown cells.

**Comment by HarryZhang1224** [comment]
Only brown cells:
![Untitled](https://github.com/user-attachments/assets/bbe4a569-2a3c-4f18-87dd-3dd367aa342f)

**Comment by kueckelj** [comment]
Okay. What is the output of `recSgsRes()`?  And whats the output of `getCoordsRange()`? Furthermore, can you send me a download link with which I can access the SPATA2 object? I don't need any meta information about the tissue but having the data and the spatial coordinates would help. I don't have as much experience with Xenium as I have with Visium. But I do have some ideas on how to solve this. It's going to need some trial and error, though. And it would be easier to try it directly instead of telling you what to do...:  jan.kueckelhaus(at)uk-erlangen.de

**Comment by HarryZhang1224** [comment]
The output of `recSgsRes()` is ``` 0.009925155 [mm]``` and `recSgsRes()` is: ```
$x 138.092200597149, 11220.7487641433 $y 113.239543274539, 5803.23528033542 ```.

I sent the download link to the SPATA2 object to your email and also put the brown cell cluster in the metadata.

Thank you so much!

**Comment by kueckelj** [comment]
Hi, I added an option to the automatic annotation functions `createGroupAnnotations()` and `addNumericAnnotations()`: `method_outline = "alphahull"`. 

Please install SPATA2 again with `devtools::install_github("kueckelj/SPATA2")`. 

obj <- createGroupAnnotations(
    object = obj, 
    grouping = "liningFLS", 
    group = "1",
    id = "liningFLS", 
    force1 = FALSE,
    overwrite = TRUE,
    use_dbscan=TRUE, # why did you set that to FALSE? 
    method_outline = "alphahull",
    min_size=0
)

The output should outline the spots better. Note, that the state of alphahull is currently "BETA". Make sure to visualize the results prior to analysis. In some cases the outlines look weird - if you encounter that, particularly with the default settings, please tell me.

**Comment by HarryZhang1224** [comment]
Thanks so much! I will try it out and let you know if I encounter any issue!
